### PR TITLE
[Backport to v0.8] Upgrade Grafana to v7.3.5

### LINKED ISSUE #1949: [Backport to v0.8] Upgrade Grafana to v7.3.5
Backport #1860 to v0.8 branch.

### PR DESCRIPTION
**Pull request by plirglo**
Upgrade Grafana to v7.3.5 - Backport to 0.8

### CODE REVIEW

File: core/src/epicli/data/common/ansible/playbooks/roles/grafana/defaults/main.yml [hunk 1]
```diff
@@ -92,5 +89,4 @@ grafana_api_keys: "{{ specification.grafana_api_keys }}"
 # The location where the keys should be stored.
 grafana_api_keys_dir: "{{ lookup('env', 'HOME') }}/grafana/keys"
 
-grafana_environment: {}
-
+grafana_environment: {}
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/grafana/defaults/main.yml]: \n

File: core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/api_keys.yml [hunk 2]
```diff
@@ -25,17 +26,18 @@
...
+    dest: "{{ grafana_api_keys_dir }}/{{ inventory_hostname }}/{{ item.item.name }}.key"
+    content: "{{ item.json.key }}"
     backup: false
-  when: item['json'] is defined
-  with_items: "{{ new_api_keys['results'] }}"
-  delegate_to: localhost
\ No newline at end of file
+    mode: u=rw,go=
+  when: item.json is defined
+  loop: "{{ new_api_keys.results }}"
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/api_keys.yml]: \n

File: core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/configure.yml [hunk 3]
```diff
@@ -59,6 +81,6 @@
 - name: Enable and start Grafana systemd unit
   systemd:
     name: grafana-server
-    enabled: true
     state: started
+    enabled: true
     daemon_reload: true
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/configure.yml]: \n

File: core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/dashboards.yml [hunk 4]
```diff
@@ -132,30 +86,28 @@
...
       when: grafana_provisioning_synced
 
     - name: Remove dashbards not present on deployer machine (synchronize)
-      become: true
       file:
         path: "{{ item }}"
         state: absent
-      with_items: "{{ _dashboards_present_list | difference( _dashboards_copied_list ) }}"
+      loop: "{{ dashboards_present_list | difference(dashboards_copied_list) }}"
       when: grafana_provisioning_synced
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/dashboards.yml]: \n

File: core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/datasources.yml [hunk 5]
```diff
@@ -26,6 +10,5 @@
     backup: false
     owner: root
     group: grafana
-    mode: 0640
-  notify: restart grafana
-  when: grafana_use_provisioning
+    mode: u=rw,g=r,o=
+  notify: restart grafana
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/datasources.yml]: \n

File: core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/install.yml [hunk 6]
```diff
@@ -1,32 +1,25 @@
...
-  when: ansible_os_family == "Debian"
\ No newline at end of file
+- name: Recursively chown /etc/grafana (fix permissions)
+  file:
+    path: /etc/grafana
+    state: directory
+    owner: root
+    group: grafana
+    mode: ug+rw,o=
+    recurse: true
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/install.yml:25]: \n

File: core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/main.yml [hunk 7]
```diff
@@ -1,86 +1,28 @@
...
-    - grafana_run
 
-- include: dashboards.yml
-  when: grafana_dashboards | length > 0 or found_dashboards.matched > 0
-  tags:
-    - grafana_configure
-    - grafana_dashboards
-    - grafana_run
\ No newline at end of file
+- include_tasks: dashboards.yml
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/main.yml:28]: \n

File: core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/notifications.yml [hunk 8]
```diff
@@ -18,6 +18,6 @@
     method: POST
     body_format: json
     body: "{{ item | to_json }}"
-  with_items: "{{ grafana_alert_notifications }}"
-  no_log: true
-  when: ((alert_notifications['json'] | selectattr("name", "equalto", item['name'])) | list) | length == 0
\ No newline at end of file
+  when: alert_notifications.json | selectattr('name', '==', item.name) | length == 0
+  loop: "{{ grafana_alert_notifications }}"
+  no_log: true
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/notifications.yml]: \n

File: core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/plugins.yml [hunk 9]
```diff
@@ -7,14 +7,12 @@
...
-  until: _plugin_install is succeeded
+  loop: "{{ grafana_plugins | difference(installed_plugins.files) }}"
+  register: result
+  until: result is success
   retries: 5
   delay: 2
-  notify:
-    - restart grafana
\ No newline at end of file
+  notify: restart grafana
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/plugins.yml]: \n

File: core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/preflight.yml [hunk 10]
```diff
@@ -70,38 +69,9 @@
...
-      -days 365
-      -nodes
-      -x509
-      -subj "/C=US/ST=NY/L=NY/O=NA/CN=localhost"
-      -keyout /etc/grafana/ssl/grafana_key.key
-      -out /etc/grafana/ssl/grafana_cert.pem
-  args:
-    creates: '/etc/grafana/ssl/grafana_cert.pem'
\ No newline at end of file
+    - not grafana_cap_net_bind_service
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/grafana/tasks/preflight.yml]: \n

File: core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/grafana.yml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+  package_facts:
+    manager: auto
+  when: ansible_facts.packages is undefined
+
+- name: Upgrade Grafana
+  when: ansible_facts.packages['grafana'] is defined
+  vars:
+    grafana_defaults:
+      # Unfortunately grafana_version in role defaults refers to the "specification" document which we cannot really use here.
+      # Next future upgrade will unhardcode this value.
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/grafana.yml:12]: do we have task for it?

File: core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/grafana.yml [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+          - "Target version: {{ grafana_defaults.grafana_version }}"
+
+    # Since we do not manage custom resources like plugins (it's up to the user), upgrading grafana by just installing new binary is considered complete here.
+    # https://grafana.com/docs/grafana/latest/installation/upgrading/
+    - name: Grafana | Upgrade
+      import_role:
+        name: grafana
+        tasks_from: install
+      when:
+        - grafana_defaults.grafana_version is version(ansible_facts.packages['grafana'][0].version, '>')
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/grafana.yml]: \n

**Review by mkyc — CHANGES_REQUESTED** [review]

File: core/src/epicli/data/common/ansible/playbooks/roles/grafana/handlers/main.yml [hunk 13]
```diff
@@ -1,27 +1,23 @@
...
   become: true
   file:
     path: "{{ grafana_data_dir }}/dashboards"
     state: directory
     recurse: false
-    mode: 0755
-  listen: "provisioned dashboards changed"
\ No newline at end of file
+    mode: u=rwx,go=rx
+  listen: provisioned dashboards changed
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/grafana/handlers/main.yml]: \n

**Review by erzetpe — CHANGES_REQUESTED** [review]

File: core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/grafana.yml (hunk 11 above)
Comment (plirglo) [reply core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/grafana.yml:12]: I will create task to refactor Grafana role

**Review by erzetpe — APPROVED** [review]

**Review by mkyc — APPROVED** [review]

**Pull request merged**